sway/sway-output.5.scd

```diff
@@ -117,37 +117,29 @@ must be separated by one space. For example:
 	(ie. blank the screen but keep workspaces as-is), one can set DPMS to off.
 
 *output* <name> max_render_time off|<msec>
-	When set to a positive number of milliseconds, enables delaying output
-	rendering to reduce latency. The rendering is delayed in such a way as
-	to leave the specified number of milliseconds before the next
-	presentation for rendering.
-
-	The output rendering normally takes place immediately after a
-	presentation (vblank, buffer flip, etc.) and the frame callbacks are
-	sent to surfaces immediately after the rendering to give surfaces the
-	most time to draw their next frame. This results in slightly below 2
-	frames of latency between the surface rendering and committing new
-	contents, and the contents being shown on screen, on average. When the
-	output rendering is delayed, the frame callbacks are sent immediately
-	after presentation, and the surfaces have a small timespan (1 /
-	(refresh rate) - max_render_time) to render and commit new contents to
-	be shown on the next presentation, resulting in below 1 frame of
-	latency.
+	Controls when sway composites the output, as a positive number of
+	milliseconds before next vblank. A smaller number leads to fresher
+	composited frames and lower perceived input latency, while a larger number
+	gives sway more time to composite and lowers the likelihood of delayed
+	frames.
```

Comment:
"fresher composited frames"  and "lower likelihood of delayed frames" doesn't seem like a good way to differentiate the two.

Reply:
It's the relevant (only?) pro and con. You lower it to get fresher frames, you increase it to avoid frames that missed vblank and got delayed till next display refresh.

Open to suggestions.

Reply:
higher input latency if too high vs dropped frames if set too low?

Reply:
For a user to see a smooth framerate, the application has to be rendering smoothly, which is outside of our control. The opposite of delayed frames is just *not* delaying frames/showing frames on time.

Sway and wayland never *drops* frames. That's up to the application. If you miss the deadline, they just end up being one frame period late.

(Maybe I'm too stubborn on the semantics - if there ends up being consensus for "dropped", we can take that instead.)

Reply:
Maybe desynchronized or inconsistent frames as alternatives? The phrasing sounded weird with the original commit, but it is fine now after reading it over.